webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/Technique.scala

```diff
@@ -191,19 +212,19 @@ object Constraint {
   }
   import NonEmptyList.one
 
-  case object NonEmpty extends  Constraint {
+  case class AllowEmpty(allow : Boolean) extends  Constraint {
```

Comment:
I'm not sure I understand why we need that? If the field can be empty, we used to not have the constraint, what changed that need that evolution? (Same for `AllowWhiteSpace` vs `NoWhiteSpace`) ?

Reply:
Since we now need to generate that constraint in the technique editor (mandatory field for technique editor), and the logic to serialize it is much simpler if we have it exlicitly https://github.com/Normation/rudder/pull/2900/files#diff-7bc55a791aa1d54ac3f6abaf79e4142cR351 (or else to set an allow_empty constraint if have to check the non presence of NonEmpty, which is kind of weird), it also simplify the parsing of the json https://github.com/Normation/rudder/pull/2900/files#diff-c54639a4587468db45968f0b40c358b5L1171 ) 

But maybe also we could adapt technique editor have a default value if not defined

Maybe I should also try with the old structure, maybe it would be ok